src/gmpv_preferences_dialog.c

```diff
@@ -173,6 +173,11 @@ static GtkWidget *build_page(	const PreferencesDialogItem *items,
 			gtk_widget_set_size_request(widget, 100, -1);
 			gtk_file_chooser_set_filename(chooser, filename);
 
+			void free_data(gpointer data, GClosure *closure)
```

Comment:
It might be a good idea to rename this to something more specific (e.g. `free_signal_data`).